packages/block-editor/src/components/link-control/link-preview.js

```diff
@@ -44,7 +44,8 @@ export default function LinkPreview( {
 
 	const displayTitle = richData?.title || value?.title || displayURL;
 
-	const isEmptyURL = ! value.url.length;
+	// url can be undefined if the href attribute is unset
+	const isEmptyURL = ! value.url?.length;
```

Comment:
In other places I also see checks for whether the `value` Is defined. Let’s replicate it for double safety.